src/kernel-install/kernel-install

```diff
@@ -90,7 +90,15 @@ if [[ ! $COMMAND ]] || [[ ! $KERNEL_VERSION ]]; then
     exit 1
 fi
 
-[ -z "$MACHINE_ID" ] && [ -f /etc/machine-id ] && [ "$(stat -fc %T /etc/machine-id)" != "tmpfs" ] && read -r MACHINE_ID < /etc/machine-id
+# Prefer to use an existing machine ID from /etc/machine-info or /etc/machine-id. If we're using the machine
+# ID /etc/machine-id, try to persist it in /etc/machine-info. If no machine ID is found, try to generate
+# a new machine ID in /etc/machine-info. If that fails, use "Default".
+
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-info ] && source /etc/machine-info && MACHINE_ID="$KERNEL_INSTALL_MACHINE_ID"
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-id ] && read -r MACHINE_ID </etc/machine-id
+[ -n "$MACHINE_ID" ] && [ -z "$KERNEL_INSTALL_MACHINE_ID" ] && echo "KERNEL_INSTALL_MACHINE_ID=$MACHINE_ID" >>/etc/machine-info
+[ -z "$MACHINE_ID" ] && echo "KERNEL_INSTALL_MACHINE_ID=$(systemd-id128 new)" >>/etc/machine-info
+[ -z "$MACHINE_ID" ] && [ -f /etc/machine-info ] && source /etc/machine-info && MACHINE_ID="$KERNEL_INSTALL_MACHINE_ID"
 [ -z "$MACHINE_ID" ] && MACHINE_ID="Default"
 
 [ -z "$BOOT_ROOT" ] && for suff in "$MACHINE_ID" "Default" "loader/entries"; do
```

Comment:
Hmm, if we are using a new KERNEL_INSTALL_MACHINE_ID, but "Default" exists, we'll use "Default" as the suffix. Shouldn't we use KERNEL_INSTALL_MACHINE_ID instead?

Reply:
Yeah, but isn't the BOOT_ROOT the most important thing here? This determines where we will actually install the stuff.

Reply:
I removed "Default" from the list of suffixes. This was changed by 9e82a74cb0f08a288f9db228a0b5bec8a7188cdb but doesn't seem right, we should only use "Default" as a suffix if no machine ID was found.